unit-tests/src/org/commcare/CommCareTestApplication.java

```diff
@@ -214,4 +221,20 @@ public ModernHttpRequester buildModernHttpRequester(Context context, URL url,
     public DataPullRequester getDataPullRequester() {
         return LocalDataPullResponseFactory.INSTANCE;
     }
+
+    @Override
+    public void afterTest(Method method) {
+        Robolectric.flushBackgroundThreadScheduler();
+        assertTrue(asyncTestsPassed);
```

Comment:
This ends up bubbling through to the end as "the failure", so it's probably worth retaining one (I know there might be more than one, but still good to have at least one) of the exceptions to this point to pass as the assertion failure. It would make the output less confusing, I think.